src/components/webview/repository.ts

```diff
@@ -97,10 +116,25 @@ export class RepositoryService {
 		return branch !== null;
 	}
 
+	public async createOrCheckoutBranch(branchName: string): Promise<void> {
+		if (this.__repo === null) {
+			return;
+		}
+
+		const branchAlreadyExists = await this.isBranchExists(branchName);
+
+		if (branchAlreadyExists) {
+			await this.__repo.checkout(branchName);
+		} else {
+			await this.__repo.createBranch(branchName, true);
+		}
+	}
+
 	public async submitChanges(
 		branchName: string,
 		remoteName: string,
 	): Promise<void> {
+		// this should throw instead returning undefined
```

Comment:
is that a TODO?

Reply:
just left a comment for myself. In  prev version these methods were throwing errors, but then we changed it to return null. I think throwing errors here makes more sense.